primitives/api/proc-macro/src/runtime_metadata.rs

```diff
@@ -0,0 +1,253 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2018-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::{parse_quote, ItemImpl, ItemTrait, Result};
+
+use crate::{
+	common::CHANGED_IN_ATTRIBUTE,
+	utils::{
+		extract_impl_trait, filter_cfg_attributes, generate_runtime_mod_name_for_trait,
+		get_doc_literals, RequireQualifiedTraitPath,
+	},
+};
+
+/// Collect extra where bounds on the parameter type.
+///
+/// `decl_runtime_apis` macro extends the generics of each trait by adding
+/// the generic `Block: BlockT`.
+///
+/// If the generic `Block` is present on the parameter type,
+/// then returns the type without any lifetimes or mutability.
+/// Otherwise, returns `None`.
+fn collect_where_bounds(ty: &syn::Type) -> Option<syn::Type> {
```

Comment:
The name is a little bit misleading :P Should be renamed :D

Reply:
Have renamed this to: `get_type_param` since that's more in-line with what it does :D